src/Kafka/Internal/Setup.hs

```diff
@@ -45,6 +46,7 @@ newtype KafkaProps = KafkaProps (Map Text Text) deriving (Show, Eq)
 newtype TopicProps = TopicProps (Map Text Text) deriving (Show, Eq)
 newtype Kafka      = Kafka RdKafkaTPtr deriving Show
 newtype TopicConf  = TopicConf RdKafkaTopicConfTPtr deriving Show
+newtype Callback = Callback (KafkaConf -> IO ())
```

Comment:
Would you mind adding a bit of documentation to help users understand what it corresponds to, and link to the callback function `setCallback`?

Reply:
I have added some documentation. The `Callback` type is now also exported from both Consumer and Producer modules, but I have kept the constructor hidden.